crypto/src/main/scala/coop/rchain/crypto/util/CertificateHelper.scala

```diff
@@ -110,14 +118,49 @@ object CertificateHelper {
     val cert = new X509CertImpl(info)
     cert.sign(privateKey, algorythm)
 
-    // Update the algorith, and resign.
+    // Update the algorithm, and resign.
     val algorithmId2 = cert.get(X509CertImpl.SIG_ALG).asInstanceOf[AlgorithmId]
     info.set(CertificateAlgorithmId.NAME + "." + CertificateAlgorithmId.ALGORITHM, algorithmId2)
     val cert2 = new X509CertImpl(info)
     cert2.sign(privateKey, algorythm)
     cert2
   }
 
+  def encodeSignatureRStoDER(signatureRS: Array[Byte]) = {
+    def toASN1Int(bytes: Array[Byte]) = new ASN1Integer(BigIntegers.fromUnsignedByteArray(bytes))
+
+    val (r, s) = signatureRS.take(64).splitAt(32)
+    val bos    = new ByteArrayOutputStream(72)
+    val seq    = new DERSequenceGenerator(bos)
+    try {
+      seq.addObject(toASN1Int(r))
+      seq.addObject(toASN1Int(s))
+      seq.close
+      bos.toByteArray
+    } catch { case _: Throwable => Array[Byte]() } finally {
```

Comment:
More nuanced exception handling please!

Reply:
@marcin-rzeznicki I'm not sure what you mean. Is there a way to use `using`?

Reply:
Since it's possible for this to throw an exception, use an error type (option or either is fine) instead of a zero-length byte array.  Ditto for the method below.

Reply:
I've added error handling to these functions.

And also found that some functions on SignatureAlg implementation can throw exceptions. I've created a new ticket for that. E.g. if DeployData#sig contains array greater than 520 bytes.
https://rchain.atlassian.net/browse/RCHAIN-3959